File: youlexuan-cart-web/src/main/webapp/js/service/payService.js
app.service('payService',function($http,$location){
    //本地支付
    this.createNative=function(){
        return $http.get('pay/createNative.do');
    }
    //查询支付状态
    this.queryPayStatus=function(out_trade_no){
        return $http.get('pay/queryPayStatus.do?out_trade_no='+out_trade_no);
    }
    //获取金额
    $scope.getMoney=function(){
        return $location.search()['money'];
    }
});

File: youlexuan-pay-interface/src/main/java/com/offcn/pay/service/PayService.java
package com.offcn.pay.service;

import java.util.*;

public interface PayService {

    /*与下单请求*/
    public Map createNative(String out_trade_no,String total_fee);
    /*查看支付状态*/
    public Map queryPayStatus(String out_trade_no);

    public Map closePay(String out_trade_no);
}
